### ISSUE TITLE
Document default Batcache behaviour in Local Environments and how to enable it

### ISSUE THREAD

**Issue by goldenapples** [humanmade/altis-cloud#332]
Action item from the question discussion is to document that Batcache is off by default on local environments and how to turn it on.

----------------------------------------------

While @stuartshields and I were investigating an issue involving batcache variants, we noticed that the default settings in altis/cloud are to [disable batcache in local environments](https://github.com/humanmade/altis-cloud/blob/master/load.php#L19) (if not in ECS or EC2). It seems easy enough to override this on a per-project by setting module config in `composer.json`, but I wondered why this setting wasn't on by default. It seems generally useful to encourage developers to test their work with page caching, and I don't see any major performance drawbacks to enabling it locally.

Would it make sure to change the default value to `true` here?

Alternatively, or in combination, it would be good to add a section to documentation (maybe in https://docs.altis-dxp.com/cloud/page-caching/) explaining how to enable or disable batcache from the Altis module config.

**Comment by roborourke** [comment]
Not sure to be honest, I generally agree that it's beneficial to have a local environment where you can encounter all the same issues you might in cloud like missing referer headers etc...

I have a feeling it was just an opinionated default config.

@joehoyle do you remember why we did this as a default for local envs?

Either way if the docs there are missing they should be added.

**Comment by joehoyle** [comment]
Mmm yeah I don't think that much thought was put into that. I would prefer to enable it in local. I think developers should be considering batcache when building. However, I see it could be an issue when wanting to tests logged out development, and you can't just refresh to get your updates.

If we switch the default I can see people continually saying "hey my stuff isn't updating in local!", and the go-to would be to say "disable page caching temporarily by updating `batcache` to `true`".

I think we should start by adding a bold `Note:` block to the page-caching docs:

Note: Page Caching is not enabled in local development by default. You should also perform site QA processes with page caching enabled. When testing custom page caching configuration, be sure to set the `altis.environments.local-server.cloud.batcache` value to `true`.

Something like that?

**Comment by rmccue** [comment]
Historically, this has been the case because page caching is super annoying for local development, but in most cases I imagine people are going to be logged in locally anyway.

One thing that might be worth considering for local at least (and maybe other environments?) is whether we should display a page cache indicator of some sort. e.g. reddit shows this caching indicator on every page:
<img width="589" alt="Screen Shot 2021-03-10 at 14 30 35" src="https://user-images.githubusercontent.com/21655/110644659-3b59d380-81ad-11eb-96b1-b330e6a5a495.png">

During development I would image something a bit more obvious would be warranted. Perhaps even a default `console.log( 'Page retrieved from Batcache at XXXX-XX-XXTXX:XX:XXZ' )` or something.

**Comment by roborourke** [comment]
That info is rendered in the page markup as a comment, but maybe we could show if it's a cache hit in the Query Monitor admin bar display.

**Comment by rmccue** [comment]
Yeah, easy to miss though if you're not actively checking the page source (and it's not exactly common to do so).

> maybe we could show if it's a cache hit in the Query Monitor admin bar display

Problem is, QM doesn't display on cached pages afaik, and won't if you're not logged in.

**Comment by goldenapples** [comment]
Yeah, given the potential for confusion among developers it probably does make sense to leave the default as off in local environments.

I like the idea of adding a note to the Page Cache docs page.
